File: application/localvision/samples/general/example4.c
     #---  Create New Class
     Object createSubclass: "ReportItem" ;
     
     #---  Define Some Properties
     ReportItem
        defineFixedProperty: 'header' .
        defineFixedProperty: 'dataType' .
        defineFixedProperty: 'width' .
        defineFixedProperty: 'accessBlock' .
     ;

     #---  Create Some Instances
     !ticker <- ReportItem new
        do: [ :header <- "Ticker" ;
              :dataType <- ^global String ;
              :width <- 10 ;
              :accessBlock <- [ ticker ] ;
            ] ;
     
     !name <- ReportItem new
        do: [ :header <- "Name" ;
              :dataType <- ^global String ;
              :width <- 25 ;
              :accessBlock <- [ name ] ;
            ] ;
     
     !sales <- ReportItem new
        do: [ :header <- "Sales" ;
              :dataType <- ^global Number ;
              :width <- 12 ;
              :accessBlock <- [ sales ] ;
            ] ;
     
     !pe <- ReportItem new
        do: [ :header <- "PE Ratio" ;
              :dataType <- ^global Number ;
              :width <- 10 ;
              :accessBlock <- [ price / earningsPerShare ] ;
            ] ;

File: application/localvision/samples/general/example4.f
     #---  Create New Class
     Object createSubclass: "Report" ;
     
     #---  Define Some Properties
     Report
        defineFixedProperty: 'itemList' .
        defineFixedProperty: 'criteria' .
        defineFixedProperty: 'sort' .
        defineFixedProperty: 'sortIsUp' .
     ;
     
     #--- Define a Report Instance
     !rep1 <- Report newAs: "Report1" ;
     rep1 :itemList <- ticker, name, sales, pe ;
     rep1 :criteria <- [ sales > 50000 ] ;
     rep1 :sort <- [ sales ] ;
     rep1 :sortIsUp <- FALSE ;
     
     #---  Define A Method to display the Report
     Report
     defineMethod:
     [ | displayFor: list | 
     
       itemList
        do: [ dataType isNumber 
                 ifTrue: [ header print: (width * -1) ] 
                ifFalse: [ header print: width ] ;
            ] ;
       newLine print ;
     
       !universe <- list select: criteria ;
       sortIsUp 
          ifTrue: [ :universe <- universe sortUp: sort ] 
         ifFalse: [ :universe <- universe sortDown:  sort ] ;
     
       universe
       do: [ !element <- ^self ; 
             ^my itemList
             do: [ ^my element send: accessBlock . print: width ] ;
             newLine print ; 
           ] ;
     ] ;
     
     #--- And run it for rep1 on testList
     rep1 displayFor: testList


